### PR TITLE
feat(node,node-wasm)!: Integrate graceful shutdown in WASM

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: cli/static/lumina_node.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Equivalent to `../../node-wasm/js/index.js` but loads `/wasm/lumina_node_wasm.js`
+
+import init, { NodeClient } from "/wasm/lumina_node_wasm.js"
+
+/**
+* Spawn a worker running lumina node and get the `NodeClient` connected to it.
+*/
+export async function spawnNode() {
+    await init();
+    let worker = new Worker(new URL("/js/worker.js", import.meta.url), { type: 'module' });
```
Comment (zvolin) [inline cli/static/lumina_node.js:10]: maybe note that in the upmost comment

**Review by zvolin — APPROVED** [review]
nice

File: node-wasm/js/worker.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
-import init, { NodeWorker, NodeClient } from "lumina-node-wasm"
+import init, { NodeWorker } from "lumina-node-wasm"
 
 Error.stackTraceLimit = 99;
 
 init().then(async () => {
-    let worker = new NodeWorker(self);
-    console.log("starting worker: ", worker);
+  let worker = new NodeWorker(self);
+  console.log("starting worker: ", worker);
```
Comment (zvolin) [inline node-wasm/js/worker.js:7]: iirc this prints some pointer, if so, maybe let's just change this to sth like
```suggestion
  console.log("Starting NodeWorker");
```

**Review by zvolin — APPROVED** [review]
here be dragons!

File: cli/src/server.rs [hunk 3]
```diff
@@ -48,8 +59,14 @@ async fn serve_index_html() -> Result<Response, StatusCode> {
 async fn serve_embedded_path<Source: RustEmbed>(
     Path(path): Path<String>,
 ) -> Result<Response, StatusCode> {
-    if let Some(content) = Source::get(&path) {
+    if let Some(mut content) = Source::get(&path) {
         let mime = mime_guess::from_path(&path).first_or_octet_stream();
+
+        if mime == mime_guess::mime::APPLICATION_JAVASCRIPT {
+            // Here be dragons!
```
[the remark this answers is not in the document]
Comment (fl0rek) [reply cli/src/server.rs:66]: :see_no_evil:

**Review by fl0rek — APPROVED** [review]
:guitar:

**Pull request merged**